assets/js/volunteer.js

```diff
@@ -0,0 +1,257 @@
+        hours: !!this.newVolunteer.hours.trim(),
+        cocaffirmation: this.newVolunteer.cocaffirmation === true
+      }
+    },
+    isValid: function () {
+      var validation = this.validation
+      return Object.keys(validation).every(function (key) {
+        return validation[key]
+      })
+    }
+  },
+  methods: {
+    onDrop: function (e) {
+      e.stopPropagation()
+      e.preventDefault()
+      var files = e.dataTransfer.files
+      this.createFile(files[0])
+    },
+    onChange(e) {
+      var files = e.target.files
+      this.createFile(files[0])
+    },
+    createFile(file) {
+      if (!file.type.match("image.*")) {
+        alert("Please select an image file")
+        return
+      }
+
+      var img = new Image()
+      var reader = new FileReader()
+      var vm = this.newVolunteer
+      vm.img = file
+
+      reader.onload = function (e) {
+        vm.photo = e.target.result
+      }
+      reader.readAsDataURL(file)
+    },
+    removeFile(e) {
+      e.stopPropagation()
+      e.preventDefault()
+      this.newVolunteer.photo = null;
+      this.newVolunteer.img = null;
+    },
+    objItem: function (name, id) {
+      return {
+        name: name,
+        id: id
+      }
+    },
+    addVolunteer: function () {
+      if (this.isValid) {
+        this.postToAPI(this.newVolunteer)
+        Object.keys(this.newVolunteer).forEach((key) => {
+          this.newVolunteer[key] = ""
+        })
+      } else {
+        console.log(
+          `invalid volunteer ${this.isValid} ${JSON.stringify(this.newVolunteer)}`
+        )
+      }
+    },
+    mapToIds: function (list) {
+      return list.map((elem) => elem.id)
+    },
+    postToAPI: function (volunteer) {
+      let vol = JSON.stringify({
+        "First Name": volunteer.firstName,
+        "Last Name": volunteer.lastName,
+        "Primary Email": volunteer.primEmail,
+        Phone: volunteer.phone,
+        "Website 1": volunteer.work,
+        "Website 2": volunteer.other,
+        City: volunteer.city,
+        State: volunteer.state,
+        LinkedIn: volunteer.linkedin,
+        Twitter: volunteer.twitter,
+        "Available hours/week": parseInt(volunteer.hours),
+        "Talent notes": volunteer.misc,
+        Skills: this.mapToIds(volunteer.chosenPositionList),
+        Passions: this.mapToIds(volunteer.chosenPassionList),
+        Activities: this.mapToIds(volunteer.chosenActivitiesList),
+        photo: volunteer.photo,
+        "I am interested in contributing my skills to ...":
+          volunteer.interested,
+        "COC Affirmation": volunteer.cocaffirmation
+      })
+
+      let config = {
+        headers: {
+          "content-type": "multipart/form-data"
+        }
+      }
+
+      axios
+        .post(
+          `https://wduc7ys73l.execute-api.us-east-1.amazonaws.com/dev/volunteers`,
+          vol,
+          config
+        )
+        .then((response) => {
+          if (response.status == "200") {
```

Comment:
Is the response status a string?

Reply:
good question. actually this is probably not needed. if there's any error, the 'catch' executes. response.status is expected to be 200 if we make it into the "then".